src/amo/components/AddonRecommendations/index.js

```diff
@@ -85,7 +86,7 @@ export class AddonRecommendationsBase extends React.Component<Props> {
     if (newRecommendations && oldRecommendations !== newRecommendations) {
       const { fallbackReason, loading, outcome } = newRecommendations;
 
-      if (loading) {
+      if (loading || errorHandler.hasError()) {
```

Comment:
Looking at this again, after looking at the new test, it seems like we should move this test to the top of the function. I.e., the first thing we do in `componentDidUpdate` is check for `errorHandler.hasError()` and if it is `true` then we `return`. We shouldn't be calling `dispatchFetchRecommendations` if there was an error.

Can you please update this, and then change the new test to check that we do not call `dispatchFetchRecommendations` if there is an error?

Reply:
But issue is that after `dispatchFetchRecommendations`,  if sagas throw error than the 500 page is shown. To handle error we should put condition after calling `dispatchFetchRecommendations`. Can you please read this comment  https://github.com/mozilla/addons-frontend/issues/6001#issue-352708353 . I think test cases are not describing the case properly. @bobsilverberg

Reply:
I think having it in either place is fine, so we can just leave it where it is.

Thanks for your work on this patch @TheSTL.